cime/tools/mapping/gen_domain_files/test_gen_domain.sh

```diff
@@ -0,0 +1,134 @@
+#!/bin/bash
+
+# get arguments
+# Need --cime_root=
+#      --inputdata_root=
+
+cime_root="default"
+inputdata_root="default"
+
+for arg in "$@"
+do
+case $arg in
+    -c=*|--cime_root=*)
+    cime_root="${arg#*=}"
+    shift
+    ;;
+
+    -i=*|--inputdata_root=*)
+    inputdata_root="${arg#*=}"
+    shift
+    ;;
+esac
+done
+if [[ ${inputdata_root} == "default" ]]; then
+    echo "Error: inputdata_root not set" >&2
+    exit 1;
+fi
+if [[ ${cime_root} == "default" ]]; then
+    echo "Error: cime_root not set" >&2
+    exit 1;
+fi
+
+# Setup the test case. gen_domain takes as inputs a mapping file and the names
+# of the ocean and land grids. Note that we could probably get these by parsing
+# the specified baseline filenames or global netcdf attributes, so that we
+# could maybe simplify this to specifying just ocn_baseline and lnd_baseline,
+# and then figuring out which mapping file and grid names we need.
+ocn_name=oQU240
+lnd_name=ne4np4
+mapping_file=${inputdata_root}/cpl/gridmaps/oQU240/map_oQU240_to_ne4np4_aave.160614.nc
```

Comment:
So, I hard-coded these into the original script for demonstration purposes. Curious what @sarich and @jgfouca have to say about this, for this tool and others that we want to add. I am happy having this just as a minimal test of functionality (i.e., the code builds and runs and gives right answers for at least one case), but I think ideally we would want to test a few different grids and make sure we are getting the right answers (different combinations of grids test the code in different ways). Any input on this? I don't have a good answer.

Reply:
Someone will have to figure out what combinations provide the best coverage.  That can be added in a future PR.

Reply:
Totally agreed. Let's keep this PR simple and choose other grids in a future PR.